simplq/src/api/auth.js

```diff
@@ -52,8 +52,8 @@ const useMakeAuthedRequest = () => {
         Authorization: await getAuthHeaderValue(auth),
       },
     });
-    const response = await authedRequest;
-    return response.data;
+
+    return data;
   };
```

Comment:
I was under the impression that these two versions are functionally equivalent. Could someone help me understand what I'm missing here 😕 ?

Reply:
> I was under the impression that these two versions are functionally equivalent. Could someone help me understand what I'm missing here  ?

@maaverik using less variables reduces complexity of the algorithm. so instead of having 2 variables authedRequest and response we an have one.

Reply:
Btw I'm getting this list of issues from [https://sonarcloud.io/project/issues?id=SimplQ_simplQ-frontend&open=AXe6u6A1mUtjMyP-a6uv&resolved=false&severities=MINOR&sinceLeakPeriod=true&types=CODE_SMELL](url) I hope it's Ok

Reply:
> @maaverik using less variables reduces complexity of the algorithm. so instead of having 2 variables authedRequest and response we an have one.

Okay, got it. I thought there was something with promise returns that I was missing. Using less variables is a good point.

Reply:
> 
> 
> Btw I'm getting this list of issues from [here](https://github.com/SimplQ/simplQ-frontend/issues/548#issuecomment-787064042) .I hope it's Ok

Please go ahead, these are all things we should have fixed before.

Reply:
@maaverik These are all code smells returned by Sonarqube.

@kabaliisa While you fix the issues, can you add a link to the corresponding sonarqube page?

Reply:
These are great change and we are also learning a lot from your PRs. :)

Reply:
> @maaverik These are all code smells returned by Sonarqube.
> 
> @kabaliisa While you fix the issues, can you add a link to the corresponding sonarqube page in the PR description?

Cool

Reply:
This code smell was about `return makeAuthedRequest` and I am pretty sure it will still `smell` after this PR.

Reply:
This change looks good to me. @mradenovic 

```
const { data } = await ...
return data;
```

is better than 
```
authedRequest = ...
response = await ...
return response.data
```

The smell about `return makeAuthedRequest`, let me check and suppress it if it's not making sense.